FIAT/quadrature_schemes.py

```diff
@@ -329,6 +329,133 @@ def _kmv_lump_scheme(ref_el, degree):
+                2.43035184285235e-2,
+                1.66312091329395e-2,
+                3.42178857644876e-2,
+                1.73480160090330e-2,
+                1.98004044953264e-2,
+            ]
+
+            alpha = alphas[0]
+            x.extend(
+                [
+                    (alpha, alpha),
+                    (0.0, alpha),
+                    (alpha, 0.0),
+                ]
+            )  # edge midpoints (class 2 points)
+            alpha = alphas[1]
+            x.extend(
+                [
+                    (1 - alpha, alpha),
+                    (alpha, 1 - alpha),
+                    (0.0, 1 - alpha),
+                    (0.0, alpha),
+                    (alpha, 0.0),
+                    (1 - alpha, 0.0),
+                ]  # edge points (1st set of class 3 points)
+            )
+            alpha = alphas[2]
+            x.extend(
+                [
+                    (1 - alpha, alpha),
+                    (alpha, 1 - alpha),
+                    (0.0, 1 - alpha),
+                    (0.0, alpha),
+                    (alpha, 0.0),
+                    (1 - alpha, 0.0),
+                ]  # edge points (2nd set of class 3 points)
+            )
+
+            beta = betas[0]
+            x.extend(
+                [
+                    (beta, beta),
+                    (1 - 2 * beta, beta),
+                    (beta, 1 - 2 * beta)
+                ]  # interior points on bisector (1st set of class 5 points)
+            )
+            beta = betas[1]
+            x.extend(
+                [
+                    (beta, beta),
+                    (1 - 2 * beta, beta),
+                    (beta, 1 - 2 * beta)
+                ]  # interior points on bisector (2nd set of class 5 points)
+            )
+            beta = betas[2]
+            x.extend(
+                [
+                    (beta, beta),
+                    (1 - 2 * beta, beta),
+                    (beta, 1 - 2 * beta)
+                ]  # interior points on bisector (3rd set of class 5 points)
+            )
+
+            gamma = gammas[0]
+            delta = deltas[0]
+            x.extend(
+                [
+                    (gamma, delta),
+                    (1 - gamma - delta, delta),
+                    (gamma, 1 - gamma - delta),
+                    (delta, gamma),
+                    (1 - gamma - delta, gamma),
+                    (delta, 1 - gamma - delta),
+                ]  # interior points (1st set of class 6 points)
+            )
+            gamma = gammas[1]
+            delta = deltas[1]
+            x.extend(
+                [
+                    (gamma, delta),
+                    (1 - gamma - delta, delta),
+                    (gamma, 1 - gamma - delta),
+                    (delta, gamma),
+                    (1 - gamma - delta, gamma),
+                    (delta, 1 - gamma - delta),
+                ]  # interior points (2st set of class 6 points)
+            )
```

Comment:
same here

Reply:
Thank you for the suggestion! I’ve modified degrees 4, 5, and 6 to use for loops for the repeated point sets to improve readability and reduce redundancy. I considered further unifying the structure for all degrees, but due to slight structural differences, a fully unified approach might make the code more complex.

For example, degrees 3, 4, 5, and 6 share the edge points symmetrically using the alpha variable and have interior points along the bisector using the beta variable. However, only degrees 5 and 6 include additional interior points determined by the gamma and delta variables

Reply:
Thanks